### PR TITLE
feat: add fulcrum chart

### PR DESCRIPTION
**Pull request by openoms**
Implements https://github.com/cculianu/Fulcrum in a chart

exposes the electrum server ports for 
* RPC
    tcp: 50001
    ssl: 50002 (used for the `readinessProbe`, `livenessProbe` )
* websocket ports:
    ws: 50003
    wss: 50004
* stats: 8080 used for the `startupProbe` and the smoketests.

Relevant documentation:
Electrum protocol: https://electrumx-spesmilo.readthedocs.io/en/latest/protocol.html

Fulcrum settings:
https://github.com/cculianu/Fulcrum/blob/master/doc/fulcrum-example-config.conf

### CODE REVIEW

File: charts/fulcrum/templates/network-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+kind: NetworkPolicy
```
Comment (krtk6160) [inline charts/fulcrum/templates/network-policy.yaml:1]: Did fulcrum not work without the network policy? Just confirming that it works

File: dev/smoketest/main.tf [hunk 2]
```diff
@@ -46,7 +46,8 @@ resource "kubernetes_pod" "smoketest" {
 
     labels = {
       "allow-to-bitcoind" : "true",
-      "allow-to-lnd" : "true"
+      "allow-to-lnd" : "true",
+      "allow-to-fulcrum" : "true"
```
Comment (krtk6160) [inline dev/smoketest/main.tf:50]: I wonder if this is required

File: ci/testflight/fulcrum/main.tf [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+data "kubernetes_secret" "bitcoin_rpcpassword" {
+  metadata {
+    name      = "bitcoind-rpcpassword"
+    namespace = "galoy-staging-bitcoin"
+  }
+}
+
+resource "kubernetes_secret" "bitcoinrpc_password" {
```
Comment (krtk6160) [inline ci/testflight/fulcrum/main.tf:25]: There's some naming inconsistency here. `bitcoinrpc_password` here vs `bitcoin_rpcpassword` in the data resource

File: ci/testflight/fulcrum/main.tf [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+data "kubernetes_secret" "network" {
+  metadata {
+    name      = "network"
+    namespace = "galoy-staging-bitcoin"
+  }
+}
+
+resource "kubernetes_secret" "network" {
```
Comment (krtk6160) [inline ci/testflight/fulcrum/main.tf:41]: Where is this secret being used?

File: charts/fulcrum/values.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  # storageClass: "-"
+  accessMode: ReadWriteOnce
+  size: 200Gi
+
+bitcoindRpcPassSecretName: bitcoind-rpcpassword
+bitcoindRpcPort: 18443
```
Comment (krtk6160) [inline charts/fulcrum/values.yaml:65]:
```suggestion
bitcoindRpcPort: 8332
```
Should be mainnet port by default

**Review by krtk6160 — COMMENTED** [review]
Have some comments

File: dev/smoketest/main.tf (hunk 2 above)
Comment (krtk6160) [reply dev/smoketest/main.tf:50]: https://github.com/GaloyMoney/charts/pull/2428 if this passes then we can remove this change

File: ci/testflight/fulcrum/main.tf (hunk 3 above)
Comment (openoms) [reply ci/testflight/fulcrum/main.tf:25]: good catch. It is similar in the testflight galoy and lnd as well. Will unify them in a separate PR.

File: ci/testflight/fulcrum/main.tf (hunk 4 above)
Comment (openoms) [reply ci/testflight/fulcrum/main.tf:41]: It isn't. Removing.

File: charts/fulcrum/templates/network-policy.yaml (hunk 1 above)
Comment (openoms) [reply charts/fulcrum/templates/network-policy.yaml]: removed in https://github.com/GaloyMoney/charts/pull/2418/commits/65ece7ebcbb5d91336b280cada54b8101d17dfab

File: dev/smoketest/main.tf (hunk 2 above)
Comment (openoms) [reply dev/smoketest/main.tf]: removed in https://github.com/GaloyMoney/charts/pull/2418/commits/65ece7ebcbb5d91336b280cada54b8101d17dfab

File: ci/testflight/fulcrum/main.tf (hunk 4 above)
Comment (openoms) [reply ci/testflight/fulcrum/main.tf]: done in https://github.com/GaloyMoney/charts/pull/2418/commits/7744447523be13e628ed0da4d3109fc4b6a3f96f

File: ci/testflight/fulcrum/main.tf (hunk 3 above)
Comment (openoms) [reply ci/testflight/fulcrum/main.tf]: done in https://github.com/GaloyMoney/charts/pull/2418/commits/7744447523be13e628ed0da4d3109fc4b6a3f96f

**Comment by openoms** [comment]
thank you for the review the review @krtk6160, all addressed and rebased

**Review by krtk6160 — APPROVED** [review]

**Pull request merged**